payload/cache/payload_id.go

```diff
@@ -34,63 +36,67 @@ const historicalPayloadIDCacheSize = 2
 // PayloadIDCache provides a mechanism to store and retrieve payload IDs based
 // on slot and parent block hash. It is designed to improve the efficiency of
 // payload ID retrieval by caching recent entries.
-type PayloadIDCache[
-	RootT ~[32]byte, SlotT ~uint64,
-] struct {
+type PayloadIDCache struct {
 	// mu protects access to the slotToStateRootToPayloadID map.
 	mu sync.RWMutex
 	// slotToStateRootToPayloadID is used for storing payload ID mappings
-	slotToStateRootToPayloadID map[SlotT]map[RootT]engineprimitives.PayloadID
+	slotToStateRootToPayloadID map[PayloadIDCacheKey]engineprimitives.PayloadID
```

Comment:
not a big deal, but what does golang do internally when hashing a struct? shouldn't be too expensive but otherwise can just create the key manually as func of (slot, root)